src/auto-evo/steps/ModifyExistingSpecies.cs

```diff
@@ -0,0 +1,327 @@
+﻿namespace AutoEvo;
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Godot;
+using Mutation = System.Tuple<MicrobeSpecies, MicrobeSpecies, RunResults.NewSpeciesType>;
+
+public class ModifyExistingSpecies : IRunStep
+{
+    public Patch Patch;
+    public SimulationCache Cache;
+
+    public ModifyExistingSpecies(Patch patch, SimulationCache cache)
+    {
+        Patch = patch;
+        Cache = cache;
+    }
+
+    public int TotalSteps => 1;
```

Comment:
This seems like a step that should *really* be able to be split into multiple ones. It is an important part of auto-evo that each step would take at most like 50-100ms but in my testing with miches some steps took dozens of seconds to complete. One immediate suggestion I have is to preserve the state of looking for good changes and just doing at most like 50 at once before returning and letting the auto-evo system call the step method again.